### PR TITLE
Feat/138 refactor main map

### PR DESCRIPTION
**Pull request by Zach13131**
Fully refactor the main page Yandex map 
Moved from package to CDN script which requires Yandex jQuery script, to plain js. 

Add file to eslint ignore and disable ts for this file, there are variables taken from CDN that ts doesn't understand and other errors that can't be resolved

Features:
- Works much faster
- Full control of anything on the map
- Fetching data dynamically for the visible part only

### CODE REVIEW

File: src/pages/main/components/MapModal.tsx [hunk 1]
```diff
@@ -1,106 +1,266 @@
...
+      checkInDate: checkInDate || null,
+      checkOutDate: checkOutDate || null,
+    };
+
+    const queryParamsString = Object.entries(newParams)
+      .filter(([key, value]) => value !== null)
+      .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(value)}`)
+      .join('&');
+
+    const url = `http://localhost:3000/api/accommodations/map?bbox=%b&${queryParamsString}`;
```
Comment (gitkoshelew) [inline src/pages/main/components/MapModal.tsx]: use url from env file, please

File: src/pages/main/components/MapModal.tsx [hunk 2]
```diff
@@ -1,106 +1,266 @@
...
+      minPeople: Number(minPeople) || null,
+      orderByPrice: orderByPrice == 'any' ? null : orderByPrice,
+      orderByPeople: orderByPeople == 'any' ? null : orderByPeople,
+      orderByRoom: orderByRoom == 'any' ? null : orderByRoom,
+      location: location || null,
+      checkInDate: checkInDate || null,
+      checkOutDate: checkOutDate || null,
+    };
+
+    const queryParamsString = Object.entries(newParams)
```
Comment (gitkoshelew) [inline src/pages/main/components/MapModal.tsx]: create util for working with params. use URL constructor from js core
let url = new URL('https://google.com/search');
url.searchParams.set('q', 'test me!');

File: src/pages/main/components/MapModal.tsx [hunk 3]
```diff
@@ -1,106 +1,266 @@
...
+      if (mapContainer && mapContainer.children.length > 0) {
+        return;
+      }
+      init(coords);
+    });
+  }
+
+  useEffect(() => {
+    if (location.trim() === '') {
+      initMapWithCoordinates([55.76, 37.64]);
```
Comment (gitkoshelew) [inline src/pages/main/components/MapModal.tsx]: as i remember we have already constant with initial coordinates, If I'm not mistaken @zilola08 knows

**Review by gitkoshelew — APPROVED** [review]

File: src/pages/main/components/MapModal.tsx (hunk 1 above)
Comment (Zach13131) [reply src/pages/main/components/MapModal.tsx]: ok, forgot that, thanks)

File: src/pages/main/components/MapModal.tsx (hunk 2 above)
Comment (Zach13131) [reply src/pages/main/components/MapModal.tsx]: done)

**Pull request merged**